module/VuFind/tests/unit-tests/src/VuFindTest/Search/Solr/ResultsTest.php

```diff
@@ -107,6 +113,230 @@ public function testGetResultTotal(): void
         $this->assertEquals(5, $results->getResultTotal());
     }
 
+    /**
+     * Test retrieving facets.
+     *
+     * @return void
+     */
+    public function testGetFacetList(): void
+    {
+        $config = $this->getMockBuilder(PluginManager::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $config->expects($this->any())
+            ->method('get')
+            ->will(
+                $this->returnValue(
+                    new Config(
+                        [
+                            'SpecialFacets' => [
+                                'hierarchical' => [
+                                    'building',
+                                ],
+                            ],
+                        ]
+                    )
+                )
+            );
+
+        $results = $this->getResultsFromResponse(
+            [
+                'response' => [
+                    'numFound' => 5
+                ],
+                'facet_counts' => [
+                    'facet_fields' => [
+                        'topic_facet' => [
+                            ['Research', 16],
+                            ['Psychotherapy', 8],
+                        ],
+                        'building' => [
+                            ['0/Main/', 11],
+                            ['1/Main/Fiction/', 5],
+                            ['0/Sub/', 2],
+                        ]
+                    ]
+                ]
+            ],
+            $this->getParams(null, $config)
+        );
+
+        // No facets configured:
+        $facets = $results->getFacetList();
+        $this->assertIsArray($facets);
+        $this->assertEmpty($facets);
+
+        // Invalid facet:
```

Comment:
Would "Facet not found in search results" be more specific than "Invalid"? "Invalid facet" makes me think that you're going to try to set some kind of really crazy value. ;-)

Reply:
Yes, changed.